deps/chakrashim/src/v8persistent.cc

```diff
@@ -61,14 +61,14 @@ template <class Callback, class Func>
 void SetObjectWeakReferenceCallbackCommon(
     JsValueRef object,
     Callback callback,
-    std::shared_ptr<WeakReferenceCallbackWrapper>* weakWrapper,
+    STD_SHARED_PTR<WeakReferenceCallbackWrapper>* weakWrapper,
     const Func& initWrapper) {
   if (callback == nullptr || object == JS_INVALID_REFERENCE) {
     return;
   }
 
   if (!*weakWrapper) {
-    *weakWrapper = std::make_shared<WeakReferenceCallbackWrapper>();
+    (*weakWrapper).reset(new WeakReferenceCallbackWrapper());
```

Comment:
Yep. That was actually what I was planning to send initially (with a custom shared_ptr impl.) `std::tr1` has a somehow decent implementation apart from c++11 `make_shared`. Maybe consider the cleanup separately as an improvement etc. ? There are `Windows only` parts with unique_ptr etc. too.